.github/workflows/maven.yml

```diff
@@ -0,0 +1,211 @@
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Set up JDK ${{ env.RELEASE_JAVA_VERSION }}
+        id: install_java
+        uses: actions/setup-java@v4
+        with:
+          distribution: zulu
+          java-version: ${{ env.RELEASE_JAVA_VERSION }}
+
+      - name: Get JNI headers
+        uses: actions/download-artifact@v4
+        with:
+          name: javah
+          path: target/native
+
+      - name: Install autotools
+        run: brew install autoconf automake libtool
+
+      - name: Build natives
+        run: resources/macos-build.sh "${{ steps.install_java.outputs.path }}" "${{ matrix.arch }}" "${{ github.workspace }}"
+
+      - name: Upload lib as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: darwin-${{ matrix.arch }}
+          path: src/main/resources/*
+
+  release:
+    name: Test including natives and release
+    needs:
+      - javah
+      - ubuntu_natives
+      - mac_natives
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Set up JDK ${{ env.RELEASE_JAVA_VERSION }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: ${{ env.RELEASE_JAVA_VERSION }}
+          cache: maven
+
+      - name: Download natives
+        uses: actions/download-artifact@v4
+        with:
+          path: target
+
+      - name: Copy natives for Maven
+        run: |
+          BASEDIR=$(pwd)/src/main/resources
+          mkdir -p $BASEDIR
+          cd target
+          for dist in */*/ ; do
+            last_dir=$(basename $dist)
+            if [[ "$last_dir" =~ ^(linux|darwin) ]]; then
+              mkdir -p "$BASEDIR/$last_dir" || true
+              cp "$dist"/*.{so,dylib} "$BASEDIR/$last_dir" || true
+            fi;
+          done;
+
+      - name: Set version
+        run: |
+          VERSION=`git describe --match "v[0-9\.]*" --long --dirty --always`
+          mvn -B versions:set -DnewVersion=${VERSION:1} -DgenerateBackupPoms=false
+          echo "VERSION=$VERSION" >> "$GITHUB_ENV"
+
+      - name: Run Maven tests with native (if PR)
+        if: github.ref != 'refs/heads/master'
+        run: mvn -B verify -DperformRelease=true
+
+      - name: Run maven tests with native (if release)
+        if: github.ref == 'refs/heads/master'
+        run: mvn -B verify
+
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v4
+
+      - name: Attach final jars (if PR)
+        uses: actions/upload-artifact@v4
+        if: github.ref != 'refs/heads/master'
+        with:
+          name: jars
+          path: target/*.jar
+
+      - name: Check out Jitsi Maven repository (if release)
+        if: github.ref == 'refs/heads/master'
+        uses: actions/checkout@v4
+        with:
+          repository: jitsi/jitsi-maven-repository
```

Comment:
This repo is a chunker. Is there a way to do a shallow clone for the extra speed?